### PR TITLE
Refactor S3 config

### PR DESCRIPTION
**Pull request by HelenMel**
__WHY__:
1) Config refactoring to support new `output.format.type` (for JSON converter) and 
2) Reduces `Add record grouper`  PR:https://github.com/aiven/aiven-kafka-connect-s3/pull/45

__WHAT__:
- Use a Common Config
- Remove Common keys 
- Use common `addOutputFieldsFormatConfigGroup` instead of `addFormatConfigGroup`

### CODE REVIEW

File: src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java [hunk 1]
```diff
@@ -88,39 +88,22 @@
...
+
+    private static final String GROUP_AWS = "AWS";
+    private static final String GROUP_FILE = "File";
+    private static final String GROUP_FORMAT = "Format";
+
     private static final Logger LOGGER = LoggerFactory.getLogger(S3SinkConfig.class);
     // FIXME since we support so far both old style and new style of property names
     //      Importance was set to medium,
     //      as soon we will migrate to new values it must be set to HIGH
     //      same for default value
```
Comment (ivanyu) [inline src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java:106]: Could you please move this comment together with the lines it was attached to?

**Review by ivanyu — CHANGES_REQUESTED** [review]
A small thingy

**Comment by ivanyu** [comment]
I'm not really sure about
> [Not exactly refactoring] Added `FILE_NAME_TEMPLATE_CONFIG` , `FILE_MAX_RECORDS` and `FILE_NAME_TIMESTAMP_TIMEZONE` as in GCS (probably, should move to a common in next release).

They will appear in the config, but will no be used. Maybe lump this part together with the PR that will make use of these configurations?

File: src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java (hunk 1 above)
Comment (HelenMel) [reply src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java:106]: Right!
- [ ] Move comment to the top

**Comment by HelenMel** [comment]
> I'm not really sure about
> 
> > [Not exactly refactoring] Added `FILE_NAME_TEMPLATE_CONFIG` , `FILE_MAX_RECORDS` and `FILE_NAME_TIMESTAMP_TIMEZONE` as in GCS (probably, should move to a common in next release).
> 
> They will appear in the config, but will no be used. Maybe lump this part together with the PR that will make use of these configurations?

Yes, Lets do this

File: src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java [hunk 2]
```diff
@@ -240,57 +224,7 @@ private static void addFileConfigGroup(final ConfigDef configDef) {
         );
     }
 
-    private static void addFormatConfigGroup(final ConfigDef configDef) {
```
Comment (HelenMel) [inline src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java:243]: use a Common config instead

File: src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java [hunk 3]
```diff
@@ -614,41 +548,6 @@ public TimestampSource getTimestampSource() {
         );
     }
 
-    public final String getConnectorName() {
```
Comment (HelenMel) [inline src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java:617]: Duplication from CommonConfig

**Review by ivanyu — APPROVED** [review]

**Pull request merged**